plugins/http2_adapter/lib/src/connection_manager_imp.dart

```diff
@@ -111,6 +115,79 @@ class _ConnectionManager implements ConnectionManager {
     return _transportState;
   }
 
+  Future<SecureSocket> _createSocket(
+    Uri target,
+    RequestOptions options,
+    ClientSetting clientConfig,
+  ) async {
+    final socket = await SecureSocket.connect(
+      target.host,
+      target.port,
+      timeout: options.connectTimeout > 0
+          ? Duration(milliseconds: options.connectTimeout)
+          : null,
+      context: clientConfig.context,
+      onBadCertificate: clientConfig.onBadCertificate,
+      supportedProtocols: ['h2'],
+    );
+
+    return socket;
+  }
+
+  Future<SecureSocket> _createSocketWithProxy(
```

Comment:
I'm not really familiar with how proxys work. Could you maybe write some comments on the general flow and why it's done like that?

Reply:
You can refer  https://www.ietf.org/rfc/rfc2817.txt point 5.2. We just create a socket(proxy tunnel), send headers and that's it, it's simple.